cmd/init.go

```diff
@@ -56,7 +58,7 @@ func Initialize(ctx context.Context, providers []string) error {
 	rootBody := f.Body()
 	requiredProviders := make([]*config.RequiredProvider, len(providers))
 	for i, p := range providers {
-		organization, providerName, provVersion, err := registry.ParseProviderNameWithVersion(p)
+		organization, providerName, provVersion, err := parseProviderCLIArg(p)
```

Comment:
I moved this function to this file and made it private. Also renamed to convey what it does